packages/portalnetwork/src/networks/networkDB.ts

```diff
@@ -149,13 +149,40 @@ export class NetworkDB {
    * Prune DB content to a new radius
    * @param radius node radius
    */
-  async prune(radius: bigint): Promise<[string, string][]> {
+  async prune(
+    radius: bigint,
+    blockIndex: Map<string, string> = new Map(),
+  ): Promise<[string, string][]> {
     const toDelete: [string, string][] = []
     for await (const [key, value] of this.db.iterator()) {
+      // Calculate distance between node and content
       const d = distance(this.nodeId, this.contentId(key))
+      // If content is out of radius -- delete content
       if (d > radius) {
-        toDelete.push([key, value])
-        await this.db.del(key)
+        // Before deleting BlockHeaderWithProof (0x00) -- Check if BlockHeaderByNumber contentKey is in radius
+        if (key.startsWith('0x00')) {
+          // First find the block number from block index
+          const blockHash = '0x' + key.slice(4)
+          const blockNumber = blockIndex.get(blockHash)!
+          const numberKey = Uint8Array.from([
+            0x03,
```

Comment:
Can we replace this with the `HistoryNetworkContentType.BlockHeaderByNumber` enum value?